src/overset/TiogaSTKIface.C

```diff
@@ -195,24 +193,46 @@ void TiogaSTKIface::reset_data_structures()
 
 void TiogaSTKIface::update_ghosting()
 {
-  uint64_t g_ghostCount = 0;
-  uint64_t nGhostLocal = elemsToGhost_.size();
-  stk::all_reduce_sum(bulk_.parallel(), &nGhostLocal, &g_ghostCount, 1);
+  stk::mesh::Ghosting* ovsetGhosting = oversetManager_.oversetGhosting_;
+  std::vector<stk::mesh::EntityKey> recvGhostsToRemove;
+
+  if (ovsetGhosting != nullptr) {
+    stk::mesh::EntityProcVec currentSendGhosts;
+    ovsetGhosting->send_list(currentSendGhosts);
+
+    sierra::nalu::compute_precise_ghosting_lists(
+      bulk_, elemsToGhost_, currentSendGhosts, recvGhostsToRemove);
+  }
+
+  size_t local[2] = {elemsToGhost_.size(), recvGhostsToRemove.size()};
+  size_t global[2] = {0, 0};
+  stk::all_reduce_sum(bulk_.parallel(), local, global, 2);
 
-  if (g_ghostCount > 0) {
+  if ((global[0] > 0) || (global[1] > 0)) {
     bulk_.modification_begin();
+    if (ovsetGhosting == nullptr) {
+      const std::string ghostName = "nalu_overset_ghosting";
+      oversetManager_.oversetGhosting_ = &(bulk_.create_ghosting(ghostName));
+    }
     bulk_.change_ghosting(
-      *(oversetManager_.oversetGhosting_), elemsToGhost_);
+      *(oversetManager_.oversetGhosting_), elemsToGhost_, recvGhostsToRemove);
     bulk_.modification_end();
 
-    sierra::nalu::populate_ghost_comm_procs(bulk_, *oversetManager_.oversetGhosting_, oversetManager_.ghostCommProcs_);
+    sierra::nalu::populate_ghost_comm_procs(
+      bulk_, *(oversetManager_.oversetGhosting_), oversetManager_.ghostCommProcs_);
 
 #if 1
     sierra::nalu::NaluEnv::self().naluOutputP0()
-      << "TIOGA: Overset algorithm will ghost " << g_ghostCount << " entities"
+      << "TIOGA: Overset algorithm will ghost " << global[0] << " elements"
```

Comment:
Might be nice to also print out how many recv-ghosts are being removed here. (global[1]).

Reply:
@alanw0 I had that initially, but it is misleading. `global[0]` only has the count of elements that are being ghosted, but `global[1]` is the count of all entities including downward relations, which causes confusion (which was once reported by Lawson for his sliding mesh runs). For example, from one of my runs with overset after a few timesteps. 

```
TIOGA: Overset algorithm will ghost 797719 entities and remove 0 entities
TIOGA: Overset algorithm will ghost 466783 entities and remove 6029750 entities
TIOGA: Overset algorithm will ghost 500647 entities and remove 6149094 entities
TIOGA: Overset algorithm will ghost 547298 entities and remove 7041817 entities
TIOGA: Overset algorithm will ghost 595677 entities and remove 8131086 entities
TIOGA: Overset algorithm will ghost 643980 entities and remove 9301716 entities
TIOGA: Overset algorithm will ghost 689529 entities and remove 10415239 entities
TIOGA: Overset algorithm will ghost 731881 entities and remove 11483473 entities
```

Reply:
@alanw0 Let me know if you still think adding _removed_ entities is useful, and I can change the code appropriately.

Reply:
That's a good point, I had forgotten that the two numbers were basically different things. I agree that adding 'removed' entities is not useful.